arraycontext/container/traversal.py

```diff
@@ -101,14 +101,14 @@ def _multimap_array_container_impl(
     """
     def rec(*_args: Any) -> Any:
         template_ary = _args[container_indices[0]]
-        assert all(
-                type(_args[i]) is type(template_ary) for i in container_indices[1:]
-                ), f"expected type '{type(template_ary).__name__}'"
-
         if (type(template_ary) is leaf_cls
                 or not is_array_container(template_ary)):
             return f(*_args)
 
+        assert all(
+                type(_args[i]) is type(template_ary) for i in container_indices[1:]
+                ), f"expected type '{type(template_ary).__name__}'"
+
```

Comment:
To reviewers: this change had to be done because in pytato leaf arrayclasses are not identical. i.e. one entry could be a `pt.IndexLambda`, other could be `pt.Placeholder``